src/EA.Weee.Web/Areas/AatfReturn/Controllers/ReturnsController.cs

```diff
@@ -33,20 +36,80 @@ public ReturnsController(Func<IWeeeClient> apiClient, BreadcrumbService breadcru
 
         [HttpGet]
         public virtual async Task<ActionResult> Index(Guid organisationId)
+        {
+            ReturnsViewModel viewModel = await PrepareViewModel(organisationId, null);
+
+            await SetBreadcrumb(organisationId, BreadCrumbConstant.AatfReturn);
+
+            return View(viewModel);
+        }
+
+        private async Task<ReturnsViewModel> PrepareViewModel(Guid organisationId, int? complianceYear, string quarter = "All")
         {
             using (var client = apiClient())
             {
-                var @return = await client.SendAsync(User.GetAccessToken(), new GetReturns(organisationId, FacilityType.Aatf));
+                ReturnsData @return = await client.SendAsync(User.GetAccessToken(), new GetReturns(organisationId, FacilityType.Aatf));
+
+                List<int> complianceYearList = @return.ReturnsList.Select(x => x.Quarter.Year).Distinct().ToList();
+                int latestComplianceYear = complianceYearList.OrderByDescending(x => x).FirstOrDefault();
+
+                ReturnsViewModel viewModel = mapper.Map<ReturnsViewModel>(@return);
 
-                var viewModel = mapper.Map<ReturnsViewModel>(@return);
+                //Need to do the below 2 lines when compliance year drop down changes, so that the quarter dropdown dynamically changes
+                List<Quarter> quartersForLatestComplianceYear = @return.ReturnsList.Select(x => x.Quarter).ToList();
+                List<string> filteredQuarterList = quartersForLatestComplianceYear.Where(x => x.Year == latestComplianceYear).Select(a => a.Q.ToString()).Distinct().ToList();
 
+                viewModel.QuarterList = filteredQuarterList;
+                viewModel.ComplianceYearList = complianceYearList;
+                
                 viewModel.OrganisationName = (await client.SendAsync(User.GetAccessToken(), new GetOrganisationInfo(organisationId))).OrganisationName;
                 viewModel.OrganisationId = organisationId;
 
+                if (complianceYear != null)
+                {
+                    viewModel.SelectedComplianceYear = complianceYear.GetValueOrDefault();
+                    viewModel.Returns = viewModel.Returns.Where(p => p.ReturnViewModel.Year == complianceYear.ToString()).ToList();
+                }
+                else
+                {
+                    viewModel.SelectedComplianceYear = latestComplianceYear;
+                }
+
+                viewModel.SelectedQuarter = quarter;
+                if (quarter != "All")
+                {
+                    viewModel.Returns = viewModel.Returns.Where(p => p.ReturnViewModel.Quarter == quarter).ToList();
+                }
+
                 await SetBreadcrumb(organisationId, BreadCrumbConstant.AatfReturn);
 
+                return viewModel;
+            }
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public virtual async Task<ActionResult> Index(ReturnsViewModel model, bool applyFilter = false)
```

Comment:
Does this need to be a post action... can we use the Index action with nullable compliance year and quarter action parameters? The anchor link can be styled to be a button... and simple link to /index?comlianceyear=blah&quarter=blah